### ISSUE TITLE
a11y violation on CvModal: Interactive component with WAI-ARIA role 'dialog' does not have a programmatically associated name

### ISSUE THREAD

**Issue by Sophy-Chhong** [carbon-design-system/carbon-components-vue#1245]
## Detailed description

> Describe in detail the issue you're having.

Running the IBM a11y-checker plugin results in a violation.

> Is this a feature request (new component, new icon), a bug, or a general issue?

bug

> Is this issue related to a specific component?

CvModal

> What did you expect to happen? What happened instead? What would you like to see changed?

There should be no violations

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

carbon vue 2.39

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

no pressure

## Additional information

<img width="1388" alt="Screenshot 2021-08-18 at 10 19 15" src="https://user-images.githubusercontent.com/89127731/129863638-4786af44-34cd-4510-a6dd-a012faa154a9.png">
